nixos/modules/config/pulseaudio.nix

```diff
@@ -160,6 +167,31 @@ in {
             if activated.
           '';
         };
+
+        resample-method = mkOption {
+          type = types.str;
+          default = "auto";
+          description = ''
+            The resampling method the pulseaudio daemon should use.
+            Possible values can be shown by <literal>pulseaudio --dump-resample-methods</literal>.
+          '';
+        };
+
+        flat-volumes = mkOption {
+          type = types.bool;
+          default = true; # TODO: is this a sane default? using the pa default for now
+          description = ''
+            Wether PulseAudio should adjust the sink volume based on playback volumes
+          '';
+        };
+
+        extraConfig = mkOption {
```

Comment:
Instead of going through plaintext splicing, this should use an attribute set and apply a generator after applying all options.
https://github.com/NixOS/nixpkgs/pull/20903